### PR TITLE
enhancement: add checkPermission in CloudMetadataClient to do some additional permission checks

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-feature

#### What this PR does / why we need it:
Add `checkPermission` method in `CloudMetadataClient` to do some additional permission checks. The default implementation will do noting.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java [hunk 1]
```diff
@@ -70,4 +85,9 @@ public interface CloudMetadataClient {
     OBTenantEndpoint getTenantEndpoint(@NotBlank String instanceId, @NotBlank String tenantId);
 
     OBDatabaseUser getSysTenantUser(@NotBlank String instanceId);
+
+    enum PermissionAction {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java:89]: RamPermissionAction?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java [hunk 2]
```diff
@@ -37,6 +40,18 @@ public interface CloudMetadataClient {
...
+     * Permission check hook. If you want to check some permission additionally, you can override this
+     * method. The default implementation will always return true.
+     * 
+     * @param tenant, the tenant to be checked
+     * @param type
+     * @return true if permission check passed, otherwise false
+     */
+    default void checkPermission(OBTenant tenant, OBInstanceType type, boolean isForAll,
+            @NotNull PermissionAction action)
+            throws AccessDeniedException {}
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java:53]: useless breaking line

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java (hunk 1 above)
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java:89]: I assume that it is not related to RAM in the interface layer. RAM is just an authentication-check implementation.

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java:89]: CloudPermissionAction, cause that PermissionAction is too common to misunderstand it.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java (hunk 2 above)
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java (hunk 1 above)
Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/connection/CloudMetadataClient.java]: > CloudPermissionAction? cause that PermissionAction is too common to misunderstand it.

okay, done~

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**